models/result.go

```diff
@@ -135,6 +136,24 @@ func (r *Result) HandleFormSubmit(details EventDetails) error {
 	return db.Save(r).Error
 }
 
+// HandleArbitraryEvent updates a Result with an arbitrary event (e.g Word document opened, secondary link clicked)
+func (r *Result) HandleArbitraryEvent(details EventDetails) error {
+
+	EventTitle := details.Payload.Get("title")
+
+	if EventTitle == "" {
+		return errors.New("No title supplied for arbitrary event")
+	}
+
+	event, err := r.createEvent(EventArbitraryEvent, details)
+	if err != nil {
+		return err
+	}
+	r.Status = EventTitle
```

Comment:
Hmmm... this is a good question that we should consider that's part of a bigger theme. That is, how much weight do we put on these events? Do we always consider the latest custom event to be the latest event in the chain?

I don't necessarily see a huge problem with that, but we'll need to think about it.

Reply:
Yeah agreed - I've been wondering how to handle this too. Should an event of 'Watched Training Video' trump 'Submitted Data' at the Status, etc. Weighting could be a good idea, almost like debugging levels.

Training Video: level-info
Clicked Link: level-warning
Enabled Macros: level-danger

It's sort of reflected at the moment in the `statuses` label properties. e.g label-danger vs label-info etc.